service/src/structure_comparer/manual_entries.py

```diff
@@ -25,48 +26,62 @@ def __init__(self) -> None:
     def entries(self) -> list[ManualEntriesMappingModel]:
         if self._data is None:
             raise NotInitialized("ManualEntries data was not initialized")
-
         return self._data.entries
 
-    def read(self, file: str | Path):
+    def read(self, file: str | Path) -> None:
         self._file = Path(file)
-        content = self._file.read_text(encoding="utf-8")
-
-        if self._file.suffix == ".json":
-            self._data = ManualEntriesModel.model_validate_json(content)
-        elif self._file.suffix == ".yaml":
-            self._data = ManualEntriesModel.model_validate(yaml.safe_load(content))
-
-    def write(self):
+        content = self._file.read_text(encoding="utf-8").strip()
+
+        if not content:
+            logger.warning(
+                f"ManualEntries file {self._file} is empty. Using default model."
+            )
+            self._data = ManualEntriesModel(entries=[])
+            return
+
+        try:
+            if self._file.suffix == ".json":
+                self._data = ManualEntriesModel.model_validate_json(content)
+            elif self._file.suffix == ".yaml":
+                parsed = yaml.safe_load(content) or {}
+                self._data = ManualEntriesModel.model_validate(parsed)
+            else:
+                raise ValueError(f"Unsupported file extension: {self._file.suffix}")
+        except Exception as e:
+            logger.exception(f"Failed to read manual entries from {self._file}: {e}")
+            raise
+
+    def write(self) -> None:
         if self._file is None:
             raise NotInitialized("ManualEntries file was not set")
-
         if self._data is None:
             raise NotInitialized("ManualEntries data was not initialized")
 
-        content = None
+        content: str | None = None
         if self._file.suffix == ".json":
             content = self._data.model_dump_json(indent=4)
         elif self._file.suffix == ".yaml":
             content = yaml.safe_dump(self._data.model_dump())
+        else:
+            raise ValueError(f"Unsupported file extension: {self._file.suffix}")
```

Comment:
I think this could crash the whole application depending from which part the write is called.